### ISSUE TITLE
Mocks are not generated not in `test` folder

### ISSUE THREAD

**Issue by nt4f04uNd** [dart-lang/mockito#429]
I'd like to mock some classes in `integration_test` folder, though codegen only works in `test` folder

**Comment by srawlins** [comment]
Does this have something to do with your `build.yaml` file? In the mockito repository itself, for example, [you can see](https://github.com/dart-lang/mockito/blob/master/build.yaml) that we generate mocks for files in `test/` and in `example/`.

**Comment by nt4f04uNd** [comment]
I don't have any `build.yaml`, do I need it necessarily for that to work?

**Comment by srawlins** [comment]
Curious, I don't think we have mockito to only do codegen on `test/` by default, but I could be wrong. If you follow the build.yaml in mockito, you can see how to specify running codegen on your `integration_test` directory.

**Comment by srawlins** [comment]
Let us know if adding  build.yaml which includes `integration_test` solves the issue.

**Comment by carlosfiori** [comment]
Hi, I'm also facing this issue.
I tried using build.yaml but doesn't work.
Here is my build.yaml:
```

targets:
  $default:
    builders:
      json_serializable:
        options:
          explicit_to_json: true
      mockito|mockBuilder:
        generate_for:
          - test/**.dart
          - test_integration/**.dart

**Comment by AlexanderJohr** [comment]
# TL;DR

Try this please:

```yaml
targets:
  $default:
    sources:
      - $package$
      - lib/$lib$
      - lib/**.dart
      - test/**.dart
      - integration_test/**.dart
    builders:
      mockito|mockBuilder:
        generate_for:
          - test/**.dart
          - integration_test/**.dart
```
# Explanation

I researched and tested in the build package source code and found the answer in the build_config [readme](https://github.com/dart-lang/build/blob/master/build_config/README.md).

The `sources` is only a subset of the files (called `sources` in the yaml file). But these files do not include the integration_test folder by default. To modify that, we can list the sources files manually. We have to include the default sources `$package$`and `lib/$lib$`, or we will get warnings if not.

By including any folder - in our case 'integration_test' it will then also be available to `generate_for`.

Here is the excerpt from the [build_config/README.md](https://github.com/dart-lang/build/blob/master/build_config/README.md):

- **generate_for**: List of String or Map, Optional:. The subset of files within
  the target's `sources` which should have this Builder applied. See `sources`
  configuration above for how to configure this.

and:

- **sources**: List of Strings or Map, Optional. The set of files within the
  package which make up this target. Files are specified using glob syntax. If a
  List of Strings is used they are considered the 'include' globs. If a Map is
  used can only have the keys `include` and `exclude`. Any file which matches
  any glob in `include` and no globs in `exclude` is considered a source of the
  target. When `include` is omitted every file is considered a match.

**Comment by siddhantkumarupmanyu** [comment]
thank @AlexanderJohr this works.

i do not want to generate the .mocks.dart file in the source so i added `build_to: cache`  but that does not work.
can you help me what i am missing?

my build.yml looks like this,
```
targets:
  $default:
    sources:
      - $package$
      - lib/$lib$
      - lib/**.dart
      - test/**.dart
      - integration_test/**.dart
    builders:
      mockito|mockBuilder:
        generate_for:
          - test/**.dart
          - integration_test/**.dart

builders:
  mockBuilder:
    import: "package:mockito/src/builder.dart"
    builder_factories: ["buildMocks"]
    build_extensions: {".dart": [".mocks.dart"]}
    build_to: cache
    auto_apply: dependents
```
